pkg/controller/cloudprivateipconfig/cloudprivateipconfig_controller.go

```diff
@@ -398,13 +396,6 @@ func (c *CloudPrivateIPConfigController) getCloudPrivateIPConfig(name string) (*
 	// for every sync, otherwise we risk acting on an old object
 	cloudPrivateIPConfig, err := c.cloudNetworkClient.CloudV1().CloudPrivateIPConfigs().Get(ctx, name, metav1.GetOptions{})
 	if err != nil {
-		if apierrors.IsNotFound(err) {
```

Comment:
This was only existing because we re-synced the final deletion (from `DeleteFunc`). If we don't re-sync the object for that event, then we don't need this check and any error is a real error